### PR TITLE
feat(ClickHouse): upgrade Docker references to 21.11.11.1 (1/2)

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Changes
This PR upgrades ClickHouse version to `21.11.11.1` for our Docker images: 

* CI
* our non-AMD64 development environments 
* [Hobby deployment](https://posthog.com/docs/self-host/deploy/hobby)

and improves as well our CH container image building process. PostHog Cloud upgrade will follow as well as for self-hosted (Helm chart). 

As part of this PR we've also found some tests that are buggy as they are relying on data being ordered while we aren't enforcing that ordering in the queries. As there are more still need to be fixed (currently breaking on ARM64) we are keeping the local development on ARM pinned to the previous CH version. I've created https://github.com/PostHog/posthog/issues/8940 to keep track of this final cleanup.

This effort is part of https://github.com/PostHog/posthog/issues/7028

I'm pushing this as ready for review but I'm still not 100% sure about it.

## How did you test this code?
* Built a new CH container image on ARM64
* CI is ✅

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -14,9 +14,7 @@ services:
         ports:
             - '6379:6379'
     clickhouse:
-        # KEEP CLICKHOUSE-SERVER VERSION IN SYNC WITH
-        # https://github.com/PostHog/charts-clickhouse/blob/main/charts/posthog/templates/clickhouse_instance.yaml#L88
```
Comment (macobo) [inline docker-compose.dev.yml:18]: I think it's worth keeping this comment

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -14,9 +14,7 @@ services:
...
-        image: yandex/clickhouse-server:${CLICKHOUSE_SERVER_IMAGE_VERSION:-21.6.5}
+        image: yandex/clickhouse-server:${CLICKHOUSE_SERVER_IMAGE_VERSION:-21.11.11.1}
```
Comment (macobo) [inline docker-compose.dev.yml:17]: How would you feel about using the same env var replacement trick elsewhere as well?

File: docker/clickhouse-builder/assets/arm64.compile.Dockerfile [hunk 3]
```diff
@@ -15,34 +37,27 @@ RUN groupadd -r clickhouse --gid=101 \
         locales \
         wget \
         tzdata \
-    && wget --progress=bar:force:noscroll "https://github.com/tianon/gosu/releases/download/$gosu_ver/gosu-$(dpkg --print-architecture)" -O /bin/gosu \
+    && wget --progress=bar:force:noscroll "https://github.com/tianon/gosu/releases/download/$GOSU_VER/gosu-$(dpkg --print-architecture)" -O /bin/gosu \
     && chmod +x /bin/gosu
 
+COPY --from=builder ClickHouse/build/programs/clickhouse* /usr/bin/
```
Comment (hazzadous) [inline docker/clickhouse-builder/assets/arm64.compile.Dockerfile:43]: What does this do with symlinks? I've got a feeling if resolves them and hence will be very large.

File: docker-compose.arm64.yml [hunk 4]
```diff
@@ -16,7 +16,7 @@ services:
     clickhouse:
         # Custom ClickHouse image for ARM64 architecture.
         # See `docker/clickhouse-builder` for more info
```
Comment (hazzadous) [inline docker-compose.arm64.yml:18]: Is this comment still relevant?

File: docker-compose.arm64.yml [hunk 5]
```diff
@@ -14,8 +14,14 @@ services:
...
-        # Built with: yarn arm64:build:clickhouse
-        image: posthog/clickhouse:v21.9.2.17-stable
+        #
+        # Custom ClickHouse image for ARM64 architecture.
+        # See the `docker/clickhouse-builder` folder for more info.
+        #
+        # Note: please keep the default version in sync across
+        #       `posthog` and the `charts-clickhouse` repos
+        #
+        image: posthog/clickhouse:${CLICKHOUSE_SERVER_IMAGE_VERSION:-v21.11.11.1-stable}
```
Comment (guidoiaquinti) [inline docker-compose.arm64.yml:24]: See https://hub.docker.com/r/posthog/clickhouse/tags

File: docker/clickhouse/users.xml [hunk 6]
```diff
@@ -1,11 +1,14 @@
 <?xml version="1.0"?>
-<yandex>
+<clickhouse>
```
[the remark this answers is not in the document]
Comment (guidoiaquinti) [reply docker/clickhouse/users.xml:2]: It was switched upstream ~6 months ago https://github.com/ClickHouse/ClickHouse/blame/master/programs/server/config.xml

**Review by hazzadous — APPROVED** [review]

**Comment by hazzadous** [comment]
plugin server tests 😢

**Comment by guidoiaquinti** [comment]
> plugin server tests 😢

Yes, still not sure why they are failing on CH 21.11 and I can't repro locally 😮‍💨

File: docker/clickhouse/users.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (guidoiaquinti) [reply docker/clickhouse/users.xml]: but I can't switch to it yet as we are still going to support the 21.9 image (that uses `yandex`) :(

**Pull request merged**